### PR TITLE
Fix tests breaking with parallel and recover flags

### PR DESCRIPTION
**Pull request by chaibhave**
refs #112 

## Reason
TMAP8 tests are failing with -p 2 --n-threads 2. 

## Design
1. Using LU in tests helps prevent exodiff errors.
2. The Interface Sorption Sievert test uses a mesh BreakMeshByBlockGenerator, which is currently incompatible with distributed meshes. This test should only run on replicated meshes.
3. Added CSVDiff and Python script tests for 1-fa.

## Impact
Fixing broken tests will build confidence in TMAP8 results with larger problems that require parallel and recovery features.

### CODE REVIEW

File: test/tests/interfacekernels/InterfaceSorptionSievert/tests [hunk 1]
```diff
@@ -7,6 +7,7 @@
     input = 'interface_sorption_Sievert.i'
     exodiff = 'interface_sorption_Sievert_out.e'
     abs_zero = 1e-7
+    mesh_mode = REPLICATED
```
Comment (simopier) [inline test/tests/interfacekernels/InterfaceSorptionSievert/tests:10]: Could you also have a comment in the input file to specify that this is limited to replicated mesh.

Could you also, as @cticenhour mentioned in private discussions: note in the test spec (as a comment) that BreakMeshByBlockGenerator currently doesn't work with DISTRIBUTED.

**Review by simopier — COMMENTED** [review]
Just a first review on the current status.

File: test/tests/interfacekernels/InterfaceSorptionSievert/tests (hunk 1 above)
Comment (chaibhave) [reply test/tests/interfacekernels/InterfaceSorptionSievert/tests:12]: Done

**Comment by chaibhave** [comment]
These tests need further improvements (which will be dealt with in the PR for #123 ), but for now this fixes the tests that were breaking in parallel runs, distributed meshes or with the recovery options.

**Comment by chaibhave** [comment]
@cticenhour can you also review this and make sure if it's sufficient to close #112 ?

File: test/tests/interfacekernels/InterfaceSorptionSievert/interface_sorption_Sievert.i [hunk 2]
```diff
@@ -2,6 +2,10 @@
 # It contains two 2D blocks separated by a continuous interface.
 # InterfaceSorptionSievert is used to enforce the Sievert law and preserve flux between the blocks.
 # Checks are performed to verify concentration conservation, sorption behavior, and flux preservation.
+# Checks are performed to verify concentration conservation, sorption behavior, and flux preservation.
```
Comment (simopier) [inline test/tests/interfacekernels/InterfaceSorptionSievert/interface_sorption_Sievert.i:5]:
```suggestion
```
This sentence is already above.

**Review by simopier — APPROVED** [review]
Looks good to me.

**Review by simopier — CHANGES_REQUESTED** [review]
Sorry, I should have waited until all comments/requested changes were resolved.

**Review by simopier — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**